### PR TITLE
FXIOS-1228 ⁃ Enhanced bottom sheet

### PR DESCRIPTION
**Pull request by nbhasin2**
This has few issues with small screen but is mostly working
- Supports landscape mode
- Full, mid, none views
- Figure out good way to show partial view on landscape mode on small screen 
- Have proper state that tells the type of view we are on for bottom sheet 

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXIOS-1228)

### CODE REVIEW

**Review by dnarcese — COMMENTED** [review]
In landscape mode, is it supposed to open to fullscreen? I am seeing it open to half screen

**Comment by nbhasin2** [comment]
> In landscape mode, is it supposed to open to fullscreen? I am seeing it open to half screen

Yeah I did it that when you open we have it half way through but only on rotation it opens fully. I can change that too.

**Comment by dnarcese** [comment]
> Yeah I did it that when you open we have it half way through but only on rotation it opens fully. I can change that too.

Ya I think it might be better if we show as much of the menu as we can.

Also, I just added a commit to fix a bug I found.

File: Client/BottomSheet/BottomSheetViewController.swift [hunk 1]
```diff
@@ -190,4 +227,24 @@ class BottomSheetViewController: UIViewController, Themeable {
...
+                print("orientation not supported")
+            }
+
+            }, completion: { (UIViewControllerTransitionCoordinatorContext) -> Void in
+                self.view.setNeedsLayout()
+                self.view.layoutIfNeeded()
+                self.containerViewController?.view.setNeedsLayout()
+        })
+    }
 }
```
Comment (dnarcese) [inline Client/BottomSheet/BottomSheetViewController.swift:250]: nit: spacing and indentation here looks off

**Review by dnarcese — APPROVED** [review]

**Pull request merged**